docs/source/ProgrammingGuide/Kokkos-and-Virtual-Functions.md

```diff
@@ -147,8 +147,8 @@ For a full working example, see [the example in the repo](https://github.com/kok
 
 ## What if I need a setter that works with host values?
 
-The first problem people run into with this is that they want to initialize some fields based on host data, with a setter which is *not* a virtual function.
-Calling this setter on the device would crash if the host data is not easily copyable on the device (e.g. for a small array).
+The first problem people run into with this is when they want to initialize some fields based on host data, using a setter which is *not* a virtual function.
+Calling this setter on the device would crash if the host data doesn't form a valid type on the device: especially, if the data type is not copyable on the device (e.g. for a pointer), or if the data instance is invalid on the device.
```

Comment:
I would prefer to leave the part post :. I think it is just confusing or a repetition. 
A pointer is a valid type on device and if it points into the right memory space, the setter is perfectly valid. Even assigning a pointer to host memory on the device is valid as the violation will only occur if the pointer is read/written to.

If you want to specify anything here, I would suggest spelling out the abstract requirements. If you want an example, you could refer to the requirements and explain what is violated